### DISCUSSION TITLE
校外用户使用虚拟机搭建的ubuntu，连接校园网并配置代理以连接Github方法

### DISCUSSION THREAD

**Discussion by Yinzexu** [physics-data/faq#44] [Show and tell] [answered]
1. 访问(https://thu.services/services/)
2. 根据《清华服务使用指北》的指示，执行
`$ sudo apt-get install openconnect`
安装完毕后再执行
`$ sudo openconnect --juniper https://sslvpn.tsinghua.edu.cn`
分别输入学号和info密码（密码不显示）即可连上校园网，此时终端显示`ESP session established with server`
3. 用ubuntu内置的firefox浏览器设置代理
打开浏览器后，点击右上角关闭按钮下的三条横线，点击拓展和主题，搜索并安装“FoxyProxy 标准版”
安装后打开“FoxyProxy 标准版”，点击左侧“添加”按钮
选择HTTP代理，代理服务器为 x.x.x.x 端口 xxxx，用户名 ******* 密码 *********（见网络学堂），填写完毕后保存并在下拉框中选中此代理服务器，并检查该代理处于“On”状态。
4. 访问Github，此时应该可以正常访问。

存在问题：一段时间后终端出现`ESP detected dead peer`，并且无法上网，可尝试重启[doge]或者每隔10分钟ping一次ping.tsinghua.edu.cn（此方法来自于THU services）

**Comment by Harry-Chen** [comment, marked as answer]
感谢！
